lib/internal/bootstrap/pre_execution.js

```diff
@@ -97,14 +100,34 @@ function patchProcessObject(expandArgv1) {
   });
   process.argv[0] = process.execPath;
 
+  let base;
   if (expandArgv1 && process.argv[1] &&
       !StringPrototypeStartsWith(process.argv[1], '-')) {
     // Expand process.argv[1] into a full path.
     const path = require('path');
     try {
       process.argv[1] = path.resolve(process.argv[1]);
+      base = path.dirname(process.argv[1]);
     } catch {}
   }
+  const implicitPackageJSONPath = getOptionValue('--package');
```

Comment:
Yeah I agree with @DerekNonGeneric probably `package-json-file` or similar?

Reply:
I'd prefer not `package-json` since the target could be named something like `module-package.test` we could go for `package-file`, does that seem fine?

Reply:
`--package-config` sounds good to me, actually.

Reply:
How about `--package-path` and then also making it optional to include the `package.json` part?

Reply:
@guybedford I'm not liking the excluding of the `package.json` since it is part of the path and we don't do searching generally these days. Additionally if we are fully explicit I wonder if we could do a inline-src like @aduh95 mentioned in https://github.com/nodejs/node/issues/37848#issuecomment-803878498 for other stuff. If we do searching having it map directly to a URL would be weirder.